code/datums/autolathe/autolathe_datums.dm

```diff
@@ -177,6 +182,10 @@
 		if(istype(O))
 			O.give_positive_attachment()
 
+	if(machine_rating || quality)
+		var/obj/O = A
+		if(istype(O))
+			O.set_quality(min(quality, machine_rating))
```

Comment:
I'm not sure what you want to do with that. You check that either `machine_rating` or `quality` is non-zero, but since you assign the min of the two, it does not change anything if only one of them in non-zero (like `machine_rating = 2` and `quality = 0` will pass the test and will assign 0).

Reply:
Would it be more performant for the condition to be min(quality, machine_rating)?

Is min() more performant than a || ?
Perhaps it would save some instances of set_quality